nifi-nar-bundles/nifi-zendesk-bundle/nifi-zendesk-processors/src/test/java/org/apache/nifi/processors/zendesk/GetZendeskTest.java

```diff
@@ -0,0 +1,276 @@
+        testRunner.setProperty(ZENDESK_RESOURCE_NAME, zendeskResource.getValue());
+        testRunner.setProperty(ZENDESK_EXPORT_METHOD_NAME, exportMethod.getValue());
+        String stateKey = zendeskResource.getValue() + exportMethod.getValue();
+        assertNull(testRunner.getStateManager().getState(CLUSTER).get(stateKey));
+
+        // when
+        testRunner.run(1);
+
+        // then
+        assertEquals(DEFAULT_CURSOR_VALUE, testRunner.getStateManager().getState(CLUSTER).get(stateKey));
+    }
+
+    @ParameterizedTest
+    @MethodSource("supportedZendeskResourcesExportMethodCombinations")
+    public void testFlowFileIsCreatedAndContentIsAddedAndFlowFileAttributeIsSet(ZendeskResource zendeskResource, ZendeskExportMethod exportMethod) throws InterruptedException {
+        // given
+        server.enqueue(new MockResponse().setResponseCode(HTTP_OK)
+            .setBody(format(RESPONSE_WITH_THREE_RECORDS_TEMPLATE, zendeskResource.getResponseFieldName())));
+        testRunner.setProperty(ZENDESK_RESOURCE_NAME, zendeskResource.getValue());
+        testRunner.setProperty(ZENDESK_EXPORT_METHOD_NAME, exportMethod.getValue());
+
+        // when
+        testRunner.run(1);
+
+        // then
+        List<MockFlowFile> flowFiles = testRunner.getFlowFilesForRelationship(REL_SUCCESS_NAME);
+        assertEquals(1, flowFiles.size());
+        MockFlowFile resultFlowFile = flowFiles.get(0);
+        assertEquals("3", resultFlowFile.getAttribute(RECORD_COUNT_ATTRIBUTE_NAME));
+        assertEquals(THREE_RECORDS, resultFlowFile.getContent());
+    }
+
+    @ParameterizedTest
+    @MethodSource("supportedZendeskResourcesExportMethodCombinations")
+    public void testNoFlowFileIsEmittedWhenZeroRecordsAreSent(ZendeskResource zendeskResource, ZendeskExportMethod exportMethod) throws InterruptedException {
+        // given
+        server.enqueue(new MockResponse().setResponseCode(HTTP_OK)
+            .setBody(format(RESPONSE_WITH_ZERO_RECORDS_TEMPLATE, zendeskResource.getResponseFieldName())));
+        testRunner.setProperty(ZENDESK_RESOURCE_NAME, zendeskResource.getValue());
+        testRunner.setProperty(ZENDESK_EXPORT_METHOD_NAME, exportMethod.getValue());
+
+        // when
+        testRunner.run(1);
+
+        // then
+        List<MockFlowFile> flowFiles = testRunner.getFlowFilesForRelationship(REL_SUCCESS_NAME);
+        assertEquals(0, flowFiles.size());
+    }
+
+    @Test
+    public void testProcessExceptionIsThrownWhenTooManyRequestResponseCodeReceived() {
+        // given
+        server.enqueue(new MockResponse().setResponseCode(HTTP_TOO_MANY_REQUESTS));
+        testRunner.setProperty(ZENDESK_RESOURCE_NAME, TICKETS.getValue());
+        testRunner.setProperty(ZENDESK_EXPORT_METHOD_NAME, CURSOR.getValue());
+
+        // when + then
+        AssertionError wrapperError = assertThrows(AssertionError.class, () -> testRunner.run(1));
```

Comment:
I tried to execute the tests locally but this seems to be failing (all others pass) could you please take a look?

Reply:
Good catch, I changed the code in the previous commit, and forgot to update the tests.

Reply:
Thanks for the update.